### ISSUE TITLE
[ABIEncoderv2] Breaking branch: ICE because "Variable headStart is 1 slot(s) too deep inside the stack."  

### ISSUE THREAD

**Issue by bshastry** [ethereum/solidity#9280]
## Description

```
pragma experimental ABIEncoderV2;
contract C {
        struct S {
                int[] b;
        }
        function f() public {
                abi.decode("", (S, byte, S, byte, S, byte, int, S, byte, S, byte, int));
        }
}
```

throws

https://github.com/ethereum/solidity/blob/a1c33249f133823476470f566753eee69cff52cd/libyul/backends/evm/AsmCodeGen.cpp#L221-L225

## Environment

- Compiler version: breaking

## Steps to Reproduce

```
$ solc <input.sol>
```

**Comment by bshastry** [comment]
Looks like https://github.com/ethereum/solidity/issues/8959 to me but keeping this issue open for now

**Comment by chriseth** [comment]
This is just a regular 'stack too deep' error because of too many arguments to the decode function. Do you want us to change the exception type so that it does not confuse the fuzzer?

**Comment by chriseth** [comment]
Ah sorry, I see you already did that in https://github.com/ethereum/solidity/pull/9281 :)

**Comment by bshastry** [comment]
> Ah sorry, I see you already did that in #9281 :)

Yeah, no worries. I guess what I wanted to ask is if stack estimation can be done at compile time so that these would be compiler errors and not failed assertions? But of course, doing it generally might be more complicated at compile time than I think. So in a way I was thinking out loud here.

**Comment by bshastry** [comment]
Fixed by #9281
